pkg/templating/manifest_generator.go

```diff
@@ -37,22 +35,54 @@ const (
 	legalCharSeparator = "-"
 )
 
+// Maximum length for UID, as we are already consuming quite a few of the remaining in templates (253-200 = 53)
+const templateUIDMaxLength int = 200
+
+func NewHandler(accessToken, threescaleURL, svcID string, fixup bool) (*Handler, []error) {
+	var errs []error
+
+	// validate that we can create a k8s label with this value
+	newSvcID, validationErrs := K8sLabelValueValidation(svcID, -1, fixup)
+	if len(validationErrs) > 0 {
+		errs = append(errs, validationErrs...)
+	}
+
+	return &Handler {
+		AccessToken: accessToken,
+		SystemURL:   threescaleURL,
+		ServiceID:   newSvcID,
+	}, errs
+}
```

Comment:
Same comment as below. I'm confused as to what should actually be expected a sa result here. If we provide `fixup == true` how do we determine that it was a success?